### ISSUE TITLE
Support specifying networks at instance create time

### ISSUE THREAD

**Issue by bahamat** [joyent/node-triton#68]

**Comment by trentm** [comment]
Showing it working against COAL:

```

# We have a public (sdc_nat) pool and a private fabric network:

$ triton networks
SHORTID   NAME               SUBNET            GATEWAY        FABRIC  VLAN  PUBLIC
4abe2ffb  My-Fabric-Network  192.168.128.0/22  192.168.128.1  true    2     false
6a51e85a  sdc_nat            -                 -              -       -     true


# Create a zone just on the private, and a bastion on both:

$ triton create -n priv0 -N My-Fabric-Network minimal-64 sample-256M -w
Creating instance priv0 (c80f8740-8ab4-4150-9281-9ffa36878dcd, minimal-64@15.3.0, sample-256M)
Created instance priv0 (c80f8740-8ab4-4150-9281-9ffa36878dcd) in 14s

$ triton create -n bastion0 -N sdc_nat,My-Fabric-Network minimal-64 sample-256M -w
Creating instance bastion0 (5f112798-e51d-43f1-8cda-baa61e1e3e9e, minimal-64@15.3.0, sample-256M)
Created instance bastion0 (5f112798-e51d-43f1-8cda-baa61e1e3e9e) in 18s

$ triton insts
SHORTID   NAME      IMG                STATE    PRIMARYIP    AGO
c80f8740  priv0     minimal-64@15.3.0  running               1m
5f112798  bastion0  minimal-64@15.3.0  running  10.88.88.33  17s


# Now SSH to the bastion, and from there to the priv0 zone:

$ triton inst -j priv0 | json ips
[
  "192.168.128.18"
]

$ triton ssh bastion0 -A
Warning: Permanently added '10.88.88.33' (RSA) to the list of known hosts.
   __        .                   .
 _|  |_      | .-. .  . .-. :--. |-
|_    _|     ;|   ||  |(.-' |  | |
  |__|   `--'  `-' `;-| `-' '  ' `-'
                   /  ; Instance (minimal-64 15.3.0)
                   `-'  https://docs.joyent.com/images/smartos/minimal

[root@5f112798-e51d-43f1-8cda-baa61e1e3e9e ~]# ping 192.168.128.18
192.168.128.18 is alive
[root@5f112798-e51d-43f1-8cda-baa61e1e3e9e ~]# ssh 192.168.128.18
The authenticity of host '192.168.128.18 (192.168.128.18)' can't be established.
RSA key fingerprint is SHA256:UcJQJ25qYa/pz7TgnNPLtKZtmnaEb45AKwAR4QLQXYU.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '192.168.128.18' (RSA) to the list of known hosts.
   __        .                   .
 _|  |_      | .-. .  . .-. :--. |-
|_    _|     ;|   ||  |(.-' |  | |
  |__|   `--'  `-' `;-| `-' '  ' `-'
                   /  ; Instance (minimal-64 15.3.0)
                   `-'  https://docs.joyent.com/images/smartos/minimal

[root@c80f8740-8ab4-4150-9281-9ffa36878dcd ~]#
```


The option is `--network,-N <comma-separated-list-of-networks>`. Can be used multiple times so you can do `triton create -N network-a -N network-b ...`.

**Comment by trentm** [comment]
Published in triton@3.6.0.
